toothpick-runtime/src/main/java/toothpick/ScopeImpl.java

```diff
@@ -436,9 +436,8 @@ private void installModule(Module module) {
       return getProvider(UN_NAMED_PROVIDER_KEY);
     }
 
-    public synchronized InternalProviderImpl<? extends T> getProvider(String bindingName) {
-      InternalProviderImpl<? extends T> internalProvider = mapNameToProvider.get(bindingName);
-      return internalProvider;
+    public InternalProviderImpl<? extends T> getProvider(String bindingName) {
```

Comment:
your implementation may not crash since it only write once. But it may crash or return unexpected value when the hashmap is resizing
http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/util/HashMap.java#390
http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/util/HashMap.java#HashMap.resize%28int%29